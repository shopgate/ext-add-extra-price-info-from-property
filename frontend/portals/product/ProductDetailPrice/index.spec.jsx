import React, { Component } from 'react';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockedStore = configureStore();

beforeEach(() => {
  jest.resetModules();
});

let mockedProductId = null;
jest.mock('@shopgate/pwa-extension-kit/connectors', () => ({
  withPageProductId: WrappedComponent => () => <WrappedComponent productId={mockedProductId} />,
}));

/**
 * Creates component with provided store state.
 * @param {Object} mockedState Mocked stage.
 * @return {ReactWrapper}
 */
const createComponent = (mockedState) => {
  /* eslint-disable global-require */
  const ProductDetailPrice = require('./index').default;
  const store = mockedStore(mockedState);
  /* eslint-enable global-require */
  return mount(
    <Provider store={store}>
      <ProductDetailPrice />
    </Provider>,
    mockRenderOptions
  );
};

const mockedStateWithExtraPriceInfo = {
  product: {
    productsById: {
      foo: {
        productData: {
          productId: 'foo',
          flags: {
            extraPriceInfo: '120 Tabletten',
          },
        },
      },
    },
  },
};

const mockedStateWithoutExtraPriceInfo = {
  product: {
    productsById: {
      foo: {
        productData: {
          productId: 'foo',
        },
      },
    },
  },
};

describe('<ProductDetailPrice />', () => {
  it('should render extraPrice', () => {
    mockedProductId = 'foo';
    const component = createComponent(mockedStateWithExtraPriceInfo);
    const ExtraPriceDiv = component.find('div');

    expect(component).toMatchSnapshot();
    expect(ExtraPriceDiv.exists()).toBe(true);
  });

  it('should not render extraPrice', () => {
    mockedProductId = null;
    const component = createComponent(mockedStateWithoutExtraPriceInfo);
    const ExtraPriceDiv = component.find('div');

    expect(component).toMatchSnapshot();
    expect(ExtraPriceDiv.exists()).toBe(false);
  });
});
