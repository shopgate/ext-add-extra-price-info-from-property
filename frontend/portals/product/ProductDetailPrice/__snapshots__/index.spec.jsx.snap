// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailPrice /> should not render extraPrice 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ProductDetailPrice)>
    <ProductDetailPrice
      dispatch={[Function]}
      extraPriceInfo={null}
    />
  </Connect(ProductDetailPrice)>
</Provider>
`;

exports[`<ProductDetailPrice /> should render extraPrice 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ProductDetailPrice)>
    <ProductDetailPrice
      dispatch={[Function]}
      extraPriceInfo="120 Tabletten"
    >
      <div
        className="css-rok1oc"
      >
        120 Tabletten
      </div>
    </ProductDetailPrice>
  </Connect(ProductDetailPrice)>
</Provider>
`;
